src/odemis/driver/technolution.py

```diff
@@ -612,7 +612,15 @@ def __init__(self, name, role, parent, **kwargs):
         clockFrequencyData = self.parent.asmApiGetCall("/scan/clock_frequency", 200)
         self.clockPeriod = model.FloatVA(1 / clockFrequencyData['frequency'], unit='s', readonly=True)
         self._shape = (8000, 8000)
-        self.resolution = model.ResolutionVA((6400, 6400), ((12*8, 12*8), (1000 * 8, 1000 * 8)),
+        # Minimum resolution is determined by:
+        #  - The shape of the MPPC detector 8 cell images per field image
+        #  - The minimal cell size (at least 10 pixels wide/tall)
+        #  - The cell size shape needs to be a whole multiple of 4 (as checked in the setter)
+        # This means that the minimum resolution needs to fulfill the conditions:
+        #          min_res / 8 > 10    and    (min_res/8)/4 is an integer
+        # Making the minimum resolution (12*8) , because 12/4 is an integer (3.0)
+        # Since the maximum cell size is 1000 (devidable by 4) the maximum resolution is (1000*8)
+        self.resolution = model.ResolutionVA((6400, 6400), ((12*8, 12*8), (1000*8, 1000*8)),
```

Comment:
Can you instantiate the detector before the scanner? If yes, you could get the shape = 8 via the parent, couldn't you. Now whenever the shape is ever changed, we need to change the value here as well, as it is now hard-coded. Ideally, it is only hard-coded in one place on the detector.
Same for _shape = (8000, 8000).

Reply:
Actually I think the detector needs to be instantiated as last. Otherwise the setters of the detector VA's won't work and some other feature rely on the scanner and descanner variables.
I agree that having one place to define these values would be better, I will think about a solution. If you come up with one please let me know.

Reply:
If the detector is depending on the scanner then indeed we cannot change the order. An option I can think of is to specify the shape = 8 on the parent class and then put it on the shape of the detector and use it here on the scanner to define the resolution. Then there is only one hard-coded place.

Reply:
For _shape, maybe just write it as `self._shape = self.resolution.range[1]`.